### ISSUE TITLE
listships groups

### ISSUE THREAD

**Issue by midspace** [midspace/Space-Engineers-Admin-script-mod#11]
Perhaps enhance the listships commands to count the number of attached grids also and include them in the results.

It's pretty cluttered already, so I wonder if it's reasonable.

**Comment by Spcemarine** [comment]
There'd be too much information, that can't be displayed via chat. It would be another situation if we put it into an missionscreen. There we can use some sort of formatting.

I wonder if we should rename /listships2 it is not quite intuitive to use a digit. I'd rather name it sth. like 'searchships', but we could also make it configurable what is shown and what is not. This way there would be no need for two commands.

**Comment by midspace** [comment]
It makes sense to move listships2 into a mission screen.
The idea is listships is a light version, whilst listships2 is a heavy version with way more detail.

**Comment by Spcemarine** [comment]
I understand what the difference is and I think both commands are quite useful, but I think the digit does not imply the difference between them. That's why I suggested to rename it. It's the same with the two versions of help. 'help2' would be confusing so I thought '/?', what is short, for a short version and '/help', that is longer, for a long version would be suitable. Maybe '/listshipdetails' would be more suitable for 'listships2'. as it is not only about displaying the ships but to display the details of them.

**Comment by midspace** [comment]
All commands support unlisted aliases.
We just need to modify the Regex to use them.
And I need to stop using the StartsWith().

**Comment by Spcemarine** [comment]
Users complain about not being able to see the first entries of /listship, /status, etc.
-> put hotlists that have more entries than lines in chat can be displayed in a mission screen.

**Comment by midspace** [comment]
The listing is unordered currently.
Am considering sorting by distance to player.

**Comment by zrisher** [comment]
Block count and large/small might be a bit more useful.

**Comment by midspace** [comment]
I agree, but I'm finding it hard to arrange more data into that list without making it harder to read.

**Comment by Spcemarine** [comment]
Just had an idea: let's rename /listships2 to /elistships what is short for extendedListships to give it a proper name. For downwards compatibility I think we need some changes in the permission system but as long as we keep the former name as an alias, we are able to do some checks to keep the old command level in the permission files.

**Comment by midspace** [comment]
yes, this works for me.

**Comment by jpcsupplies** [comment]
Just a thought here - from the perspective of an end user - why not merge both commands.  If you get less than a screen full of results, display it in chat, if you get more display it in the mission box? 90% of the time i previously used "/listships2", and 10% of the time I would use "/listship <search criteria>" to find a particular thing or things..  and i only used it that 10% of the time as i felt guilty for using listship2 all the time.. plus I wanted the id to remain on screen long enough to /tps to it;  if the concern is screen clutter - they can use /id <ship number> to get extended info about that ship.   
When using listship2 previously I was mainly interested in speed/distance/location/size to find wrecks to remove, or spot spawned explore mode items, and i used ID to find out who it was before i deleted it anyway.  May as well make it work the way people actually use it?

**Comment by Spcemarine** [comment]
With your suggestion of merging the commands I had a great idea. If it doesn't get too cluttered we could make the additional information configurable for each command execution. So that you just type in `/listships` -d -l (d = distance, l = location) and you'll get a list with distances and locations of the ships. Other information like speed (-s), block count (-b; also we could specify blocktypes with "-b:refinery" or something...), etc. could also easily be displayed. Also we could sort the list regarding the first optional parameter that was given. Like, if I type `/listships -d` I get a list sorted by distance and besides the standard information (list id, entity id, name) only shows the distance to the player.

**Comment by jpcsupplies** [comment]
sorting hadn't even occurred to me, but that would be excellent, the assumption is derelicts are usually drifting, so being able to examine the furthest(or fastest moving) item first is brilliant; (just remember to update the user guide about the -switches especially if you do combine the commands.)   could even add a -o owner information filter/sort too, although that might be tricky on multi-owner builds.

**Comment by zrisher** [comment]
I'm definitely a big supporter of making these lists more informative and navigable, the information on them is really useful.
